kycdao/smart-contracts
# Test coverage

Fairly sizeable PR as it includes a whole bunch of new tests which improve test coverage quite a bit.

The following tests are added:
* Check role on authMint
* Check code already used in authMint
* Check valid use of sendGasOnAuth, also setSendGasOnAuth
* Check version()
* Check tokenVerificationURI, also set
* Check setBaseURI
* Check all 4 tokenId funs with non-existent tokens
* Check setRevoke and updateExpiry for role and non-existent token
* Check role on authorizeUpgrade, by upgrading from another acct
* Check supportsInterface, for both yes and no
* When sending eth the contract’s .call fails
    * Refunds
GSN
* Set trusted forwarder
    * Bad role
    * Actually sets forwarder, how to check?
* Check _msgData - buh bow (it's internal and we don't use it.... :()

Also a bunch of more advanced tests which uses TestWallets.sol when the receiver is a contract, such as re-entrancy e.t.c.

NB:- There are 2 small PRs which are outstanding, they should be merged to get an idea of what's only on this branch.
